packages/block-editor/src/components/date-format-control/index.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import { uniq } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { _x, sprintf, __ } from '@wordpress/i18n';
+import { dateI18n } from '@wordpress/date';
+import { useState, createInterpolateElement } from '@wordpress/element';
+import {
+	SelectControl,
+	TextControl,
+	ExternalLink,
+} from '@wordpress/components';
+
+/**
+ * So that we can illustrate the different formats in the dropdown properly,
+ * show a date that has a day greater than 12 and a month with more than three
+ * letters. Here we're using 2022-01-25 which is when WordPress 5.9 was
+ * released.
+ */
+const EXAMPLE_DATE = new Date( 2022, 0, 25 );
+
+/**
+ * The `DateFormatControl` component renders a dropdown that lets the user
+ * choose a _date format_. That is, how they want their dates to be formatted.
+ *
+ * @see https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/date-format-control/README.md
+ *
+ * @param {Object}                     props
+ * @param {string|null}                props.format     The selected date
+ *                                                      format.
+ * @param {string}                     props.siteFormat The site's date format,
+ *                                                      used to show what the
+ *                                                      date will look like if
+ *                                                      user selects _Site
+ *                                                      default_.
+ * @param {( format: string ) => void} props.onChange   Called when a selection
+ *                                                      is made.
+ */
+export default function DateFormatControl( { format, siteFormat, onChange } ) {
+	// Suggest a short format, medium format, long format, and a standardised
+	// (YYYY-MM-DD) format. The short, medium, and long formats are localised as
+	// different languages have different ways of writing these. For example, 'F
+	// j, Y' (April 20, 2022) in American English (en_US) is 'j. F Y' (20. April
+	// 2022) in German (de). The resultant array is de-duplicated as some
+	// languages will use the same format string for short, medium, and long
+	// formats.
+	const suggestedFormats = uniq( [
+		_x( 'n/j/Y', 'short date format' ),
+		_x( 'M j, Y', 'medium date format' ),
+		_x( 'F j, Y', 'long date format' ),
+		_x( 'l, F j, Y', 'full date format' ),
+		_x( 'n/j/Y g:i A', 'short date format with time' ),
+		_x( 'M j, Y g:i A', 'medium date format with time' ),
+		_x( 'F j, Y g:i A', 'long date format with time' ),
+		'Y-m-d',
```

Comment:
A question about the order:

I see the more elaborate datetime formats are grouped.

In no way a biggie, but would it make sense to bump the shorthand format at the bottom `YYYY-MM-DD` up a bit so it sits cosy beneath its slashed cousin? 

That is, from


<img width="278" alt="Screen Shot 2022-03-10 at 11 07 37 am" src="https://user-images.githubusercontent.com/6458278/157563179-3018f132-3227-4faa-98fd-ff2113724c36.png">

to 

<img width="277" alt="Screen Shot 2022-03-10 at 11 14 36 am" src="https://user-images.githubusercontent.com/6458278/157563171-945410b3-aacc-43bf-83d5-70248b63edf8.png">

Reply:
I played around with the order a little bit in ba74a6b3d0f54f15af939ac0aa9e87cdbf0930f9. I think it makes more sense now.